src/justfile.rs

```diff
@@ -253,9 +254,32 @@ impl<'src> Justfile<'src> {
       search,
     };
 
-    let mut ran = BTreeSet::new();
-    for (recipe, arguments) in grouped {
-      Self::run_recipe(&context, recipe, arguments, &dotenv, search, &mut ran)?;
+    let ran = Arc::new(Mutex::new(BTreeSet::new()));
+
+    let (tx, rx) = std::sync::mpsc::channel();
+
+    std::thread::scope(|scope| {
+      for (recipe, arguments) in grouped {
+        let run = || {
+          if let Err(err) = Self::run_recipe(&context, recipe, arguments, &dotenv, search, &ran) {
+            let _ = tx.send(err);
+            true
+          } else {
+            false
+          }
+        };
+        if config.parallel {
+          scope.spawn(run);
+        } else if run() {
```

Comment:
nit: I found it confusing when reading the code to expect `true` for the error case, but that might just be me.